src/tests.rs

```diff
@@ -801,3 +801,30 @@ fn test_slice_as_contiguous() {
     );
     assert_eq!(sv.slice(5..7).as_contiguous(), Some([5, 6].as_slice()));
 }
+
+#[test]
+fn test_extend_from_slice() {
+    let mut elements = [0u32; 100];
+    elements.try_fill(&mut rand::thread_rng()).unwrap();
+
+    // single iter
+    let mut sv: SegVec<u32> = SegVec::new();
+    sv.extend_from_slice(elements.as_slice());
+
+    assert_eq!(elements.len(), sv.len());
+    for (a, b) in elements.iter().zip(sv.iter()) {
+        assert_eq!(a, b);
+    }
+
+    // multiple iterations
+    let mut sv: SegVec<u32> = SegVec::new();
+
+    for _ in 0..8 {
+        sv.extend_from_slice(elements.as_slice());
+    }
+
+    assert_eq!(8 * elements.len(), sv.len());
+    for (a, b) in elements.iter().zip(sv.iter()) {
+        assert_eq!(a, b);
+    }
```

Comment:
I think we probably want to validate all the elements of `sv` here. My reading of this code is that we're just validating the first `N` items, where `N` is the length of `elements`. You could do something like:

```suggestion
    for (i, v) in sv.iter().enumerate() {
        assert_eq!(v, &elements[i%elements.len()]);
    }
```

Reply:
oops, you're right! totally slipped past me